x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/AbstractBlockBuilder.java

```diff
@@ -173,8 +169,10 @@ protected void adjustBreaker(long deltaBytes) {
     private void setFirstValue(int position, int value) {
         if (position >= firstValueIndexes.length) {
             final int currentSize = firstValueIndexes.length;
-            adjustBreaker((long) (position + 1 - currentSize) * Integer.BYTES);
-            firstValueIndexes = Arrays.copyOf(firstValueIndexes, position + 1);
+            final int newLength = ArrayUtil.oversize(position + 1, Integer.BYTES);
+            adjustBreaker((long) newLength * Integer.BYTES);
+            firstValueIndexes = ArrayUtil.growExact(firstValueIndexes, newLength);
+            adjustBreaker(-(long) currentSize * Integer.BYTES);
```

Comment:
It's worth saying that we could grow the `values` array at the same rate as `firstValues` but that'd be a bit more complex. And maybe less efficient. But maybe more? I have no idea. Either way, this is fine. But you did make a choice to get here.

Reply:
I added a comment in https://github.com/elastic/elasticsearch/pull/106934/commits/bc12f69ae759f2a5b32b8d10cf2bd46de902fa81